TodoApp.ConsoleApp/UI/UserInput.cs

```diff
@@ -35,6 +37,10 @@ public static string ReadText(string question, bool required = false, string def
         public static int ReadOption(string question, string[] availableOptions, bool required = false, int defaultValue = 0)
         {
             Console.WriteLine(question);
+            if (availableOptions == null)
```

Comment:
Move this validation at the top of the method.

`ArgumentNullException` requires the argument name as first parameter. use `nameof` operator to provide it.